core/src/main/scala/chisel3/experimental/Probe.scala

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.experimental
+
+import chisel3._
+import chisel3.Data.ProbeInfo
+import chisel3.internal.Builder
+import chisel3.internal.Builder.pushCommand
+import chisel3.internal.firrtl._
+
+// import firrtl.ir._
+
+/** Utilities for creating and working with probe types.
+  */
+object Probe {
+
+  /** Create a probe type from a Chisel type. This is not used for
+    * hardware.
+    */
+  private def apply_impl[T <: Data](source: => T, writable: Boolean): T = {
+    val prevId = Builder.idGen.value
```

Comment:
This does not need Scala doc as it is the macro inserted apply method. Also I'm shocked that it's allowing this to be private (since the macro should be injecting it into user code), are you sure this is being called in the tests?